core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassMonitor.scala

```diff
@@ -0,0 +1,101 @@
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{DataAccess, ReadWriteAction}
+import org.broadinstitute.dsde.rawls.fastpass.FastPassMonitor.DeleteExpiredGrants
+import org.broadinstitute.dsde.rawls.model.{FastPassGrant, Workspace}
+import org.broadinstitute.dsde.workbench.google.{GoogleIamDAO, GoogleStorageDAO}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.iam.IamResourceTypes
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import slick.dbio.DBIO
+
+import scala.concurrent.Future
+import scala.language.postfixOps
+
+object FastPassMonitor {
+  sealed trait FastPassMonitorMessage
+  case object DeleteExpiredGrants extends FastPassMonitorMessage
+  def props(dataSource: SlickDataSource, googleIamDao: GoogleIamDAO, googleStorageDao: GoogleStorageDAO)(implicit
+    openTelemetry: OpenTelemetryMetrics[IO]
+  ): Props = Props(
+    new FastPassMonitor(dataSource, googleIamDao, googleStorageDao)
+  )
+}
+
+class FastPassMonitor private (dataSource: SlickDataSource,
+                               googleIamDao: GoogleIamDAO,
+                               googleStorageDao: GoogleStorageDAO
+)(implicit openTelemetry: OpenTelemetryMetrics[IO])
+    extends Actor
+    with LazyLogging {
+  import context.dispatcher
+
+  private val openTelemetryTags: Map[String, String] = Map("service" -> "FastPassMonitor")
+  override def receive: Receive = { case DeleteExpiredGrants =>
+    openTelemetry.incrementCounter("fastpass-monitor-receive-msg", tags = openTelemetryTags).unsafeToFuture()
+    deleteExpiredGrants()
+  }
+
+  private def deleteExpiredGrants(): Future[ReadWriteAction[Unit]] =
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.fastPassGrantQuery
+        .findExpiredFastPassGrants()
+        .flatMap { expiredGrants =>
+          logger.info(s"Found ${expiredGrants.size} total expired grants")
+          // Convert to stream and group by workspaceId
+          DBIO.sequence(expiredGrants.to(LazyList).groupBy(_.workspaceId).map { case (workspaceId, workspaceGrants) =>
+            logger.info(s"Found ${workspaceGrants.size} expired grants for workspace $workspaceId")
+            // Query workspace info then group by accountEmail
+            dataAccess.workspaceQuery.findByIdOrFail(workspaceId).flatMap { workspace =>
+              DBIO.sequence(workspaceGrants.groupBy(_.accountEmail).map { case (_, accountEmailGrants) =>
+                // Remove grants for a given workspace and accountEmail
+                removeGrantsForAccountEmailInWorkspace(dataAccess, workspace, accountEmailGrants)
+              })
+            }
+          })
+        }
+        .map(_ => DBIO.successful(()))
+    }
+
+  private def removeGrantsForAccountEmailInWorkspace(dataAccess: DataAccess,
```

Comment:
I'm a little confused as to how all the `DBIO` calls are all getting executed in this function. It looks like their being called, but not actually returned by the function.

Reply:
They are the last line in the block so they are getting returned